### PR TITLE
added new controller for creating custom ExtClient

### PR DESCRIPTION
**Pull request by k4s0**
Add the possibility to create custom clientid #1453

### CODE REVIEW

File: controllers/ext_client.go [hunk 1]
```diff
@@ -248,9 +249,9 @@ func createExtClient(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 
 	var params = mux.Vars(r)
-
 	networkName := params["network"]
 	nodeid := params["nodeid"]
+	clientid := params["clientid"]
```
Comment (mattkasun) [inline controllers/ext_client.go:254]: client id should be retrieved from struct passed in request.Body not from url parameter

**Review by mattkasun — CHANGES_REQUESTED** [review]

File: controllers/ext_client.go (hunk 1 above)
Comment (k4s0) [reply controllers/ext_client.go:254]: OK but i need to use the ExtClient struct or create a new struct for this type of request?

Comment (mattkasun) [reply controllers/ext_client.go:254]: you could do either ... use the ExtClient struct but only send the ClientID param in the body of the request or use a new struct.
I would create an anonymous struct consisting of just one element and decode the body into that struct and then copy to extclient struct.  
eg.

request := struct {
     id string
}
if err := json.NewDecoder(r.body).Decode(&request); err != nil {
   //log error
} else {
   extclient.ClientID = request.id
}

Comment (k4s0) [reply controllers/ext_client.go]: I have made the changes

File: controllers/ext_client.go [hunk 2]
```diff
@@ -261,6 +261,16 @@ func createExtClient(w http.ResponseWriter, r *http.Request) {
 	}
 
 	var extclient models.ExtClient
+	var CustomExtClient models.CustomExtClient
+	
+	err := json.NewDecoder(r.Body).Decode(&CustomExtClient);
+
+	if err != nil {
+		logger.Log(1, "error creating CustomExtClient"+err.Error())
```
Comment (0xdcarns) [inline controllers/ext_client.go:269]: log message suggests an error occurred, but it's more of an absence of a clientid, would either log that no clientid was provided and one will be created, or just check if err == nil { extclient.ClientID = CustomExtClient.ClientID } (no log)

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: controllers/ext_client.go (hunk 2 above)
Comment (k4s0) [reply controllers/ext_client.go]: @0xdcarns ok you're right i have made the changes

**Review by mattkasun — APPROVED** [review]

**Comment by afeiszli** [comment]
@k4s0 is 'lorenzo' also you in the commit? Both are showing up as commiters so CLA Assistant is looking for both.

Once that is sorted we will merge.

**Comment by k4s0** [comment]
> @k4s0 is 'lorenzo' also you in the commit? Both are showing up as commiters so CLA Assistant is looking for both.
> 
> Once that is sorted we will merge.

Yes both mine because i used two different git config, in two different machine, but just only one is a github users, so how can i resolve this?

**Review by afeiszli — APPROVED** [review]

**Review by 0xdcarns — APPROVED** [review]

**Pull request merged**